tree: allow resolution of spatial builtins with "public" schema

otan opened the pull request:
Release note (sql change): Add the ability to resolve the spatial-backed
builtins in the public schema. For example, `public.st_x` works the same
as `st_x`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52983)
<!-- Reviewable:end -->

rohany reviewed (approved):
heh, we'll have to revisit this once we support user defined functions

otan commented:
better stay to remember, rethink and implement around this gotcha :)

otan committed:
tree: allow resolution of spatial builtins with "public" schema

Release note (sql change): Add the ability to resolve the spatial-backed
builtins in the public schema. For example, `public.st_x` works the same
as `st_x`.

otan commented:
bors r=rohany
